src/hotspot/share/oops/oop.cpp

```diff
@@ -119,10 +119,11 @@ bool oopDesc::is_oop(oop obj, bool ignore_mark_word) {
   }
 
   // Header verification: the mark is typically non-zero. If we're
-  // at a safepoint, it must not be zero.
+  // at a safepoint, it must not be zero. fast-locking does allow the
+  // mark to be zero at a safepoint.
   // Outside of a safepoint, the header could be changing (for example,
   // another thread could be inflating a lock on this object).
-  if (ignore_mark_word) {
+  if (ignore_mark_word || UseFastLocking) {
```

Comment:
Not clear why UseFastLocking appears here instead of in the return expression - especially given the comment above.

Reply:
I have the same question.  Based on the comment, I would expect
`return !SafepointSynchronize::is_at_safepoint() || UseFastLocking`;